src/cryptojwt/key_bundle.py

```diff
@@ -507,34 +508,35 @@ def update(self):
         :return: True if update was ok or False if we encountered an error during update.
         """
         if self.source:
-            _old_keys = self._keys  # just in case
+            with threading.Lock():
```

Comment:
You need a global variable with the lock. This will create a new Lock() for each call.

Reply:
Fix in c9e59bc3d606d4ef421417e3322b0d9801657c90, will address the lookup while updating issue as well